### PR TITLE
ci: check for typos using typos-cli

### PR DESCRIPTION
**Pull request by pacmancoder**
I frequently fail to remember to run `typos-cli` manually, so I added it in the CI.

Installed `typos-cli` binary should be cached, so overall CI speed should not change much (on GitHub runners).

### CODE REVIEW

**Comment by pacmancoder** [comment]
Aha, after merge one more false-positive should be fixed :)

**Comment by pacmancoder** [comment]
> Before we merge: a new "typos" job should be added to the CI:

Silly me, I only tested it with `xtask ci` 🤦‍♂️

**Comment by pacmancoder** [comment]
@CBenoit Done, ready for re-review

File: .github/workflows/ci.yml
Comment (CBenoit) [inline .github/workflows/ci.yml:44]: Suggestion: let’s cache the binaries, as done in other jobs:
```yml
      - name: Binary cache
        uses: actions/cache@v3
        with:
          path: ./.cargo/local_root/bin
          key: ${{ runner.os }}-bin-${{ github.job }}-${{ hashFiles('xtask/src/bin_version.rs') }}
```

**Review by CBenoit — APPROVED** [review]
Approving so you can merge after the last few details are addressed :slightly_smiling_face: 

Note: I renamed the PR title to match the conventional commit style that we already mostly follow correctly. Make sure to keep it when squashing.

File: .github/workflows/ci.yml
Comment (pacmancoder) [reply .github/workflows/ci.yml:44]: I am all over the place today 🤦‍♂️

File: xtask/src/main.rs
Comment (pacmancoder) [reply xtask/src/main.rs:70]: I think we need to keep this consistent with the code below -- we also call `web::install(&sh)?;` here to run `web::check(&sh)?;` later, so lets keep install for `check` too

File: crates/ironrdp-graphics/src/zgfx/mod.rs
Comment (CBenoit) [inline crates/ironrdp-graphics/src/zgfx/mod.rs:581]: Actually, it seems this is not a typo in the comment, the payload on the left appears to be indeed holding "fox jumps ove" without "r" :joy:

**Pull request merged**
